components/eamxx/src/share/util/scream_family_tracking.hpp

```diff
@@ -49,17 +49,17 @@ class FamilyTracking : public ekat::enable_shared_from_this<DerivedType>
   FamilyTracking& operator= (const tracking_type&) = default;
   ~FamilyTracking ();
 
-  void create_parent_child_link (const std::weak_ptr<derived_type>& parent);
+  void create_parent_child_link (const std::shared_ptr<derived_type>& parent);
 
-  std::weak_ptr<derived_type> get_parent () const { return m_parent; }
+  std::shared_ptr<derived_type> get_parent () const { return m_parent; }
 
   const std::list<std::weak_ptr<derived_type>>& get_children () const { return m_children; }
 protected:
 
   // Check if a weak_ptr points to the same object as this class
   bool is_same (const std::weak_ptr<derived_type>& src) const;
 
-  std::weak_ptr<derived_type>              m_parent;
+  std::shared_ptr<derived_type>            m_parent;
   std::list<std::weak_ptr<derived_type>>   m_children;
```

Comment:
Note: must keep children as weak_ptr, or else we create a cyclic shared_ptr dependency that makes it so that the objects are never deleted.